AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/ui/adapter/AppRecyclerViewAdapter.java

```diff
@@ -0,0 +1,111 @@
+package com.example.appactionvisualizer.ui.adapter;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.graphics.drawable.Drawable;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import com.example.appactionvisualizer.R;
+import com.example.appactionvisualizer.constants.Constant;
+import com.example.appactionvisualizer.databean.ActionType;
+import com.example.appactionvisualizer.databean.AppAction;
+import com.example.appactionvisualizer.ui.activity.ActionActivity;
+import com.example.appactionvisualizer.utils.Utils;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+
+/**
+ * Adapter of AppFragment Recyclerview
+ */
+public class AppRecyclerViewAdapter extends RecyclerView.Adapter<AppRecyclerViewAdapter.ViewHolder> {
+
+  private ArrayList<AppAction> appActionArrayList;
+  private Context context;
+
+  public AppRecyclerViewAdapter(ActionType type, Context context) {
+    this.context = context;
+    if(type != null)
```

Comment:
nitpicking: in Google, we prefer to not omit the curly braces

Reply:
fixed